[base-ui][NumberInput] Support adornments (#38900)
mui/material-ui: docs/data/base/components/number-input/number-input.md

```jsx
<NumberInput min={0} step={1} shiftMultiplier={5} />
```

### Adornments

> You can use the `startAdornment` and `endAdornment` props to add a prefix or suffix to a number input.

Reviewer: ```suggestion
You can use the `startAdornment` and `endAdornment` props to add a prefix or suffix, respectively, to a Number Input.
```

> A common use case for adornments is when the component receives a specific unit of measure (like weight or currency).

Reviewer 2: ```suggestion
A common use case for adornments is displaying a specific unit of measure, like weight or currency.
```

@samuelsycamore Do you think we could simplify this sentence like this?

Reviewer: Here's how I'd phrase this:

```suggestion
Adornments can be useful for displaying units of measure, like weight or currency, alongside values.
```
